payjoin/src/receive/v1/mod.rs

```diff
@@ -270,40 +270,40 @@ impl WantsOutputs {
     pub fn substitute_receiver_script(
         self,
         output_script: &Script,
-    ) -> Result<WantsOutputs, OutputSubstitutionError> {
+    ) -> Result<Self, OutputSubstitutionError> {
         let output_value = self.original_psbt.unsigned_tx.output[self.change_vout].value;
-        let outputs = vec![TxOut { value: output_value, script_pubkey: output_script.into() }];
-        self.replace_receiver_outputs(outputs, output_script)
+        let outputs = [TxOut { value: output_value, script_pubkey: output_script.into() }];
+        self.replace_receiver_outputs(outputs.iter(), output_script)
     }
 
     /// Replace **all** receiver outputs with one or more provided outputs.
     /// The drain script specifies which address to *drain* coins to. An output corresponding to
     /// that address must be included in `replacement_outputs`. The value of that output may be
     /// increased or decreased depending on the receiver's input contributions and whether the
     /// receiver needs to pay for additional miner fees (e.g. in the case of adding many outputs).
-    pub fn replace_receiver_outputs(
+    pub fn replace_receiver_outputs<'a>(
         self,
-        replacement_outputs: Vec<TxOut>,
+        replacement_outputs: impl Iterator<Item = &'a TxOut>,
```

Comment:
I think this is mostly good except that it ought to use `IntoIterator` for max flexibility vs `Iterator`, and match interfaces like rust std libs' collect and extend() and Vec::from_iter impls.